src/injected/utils/helpers.js

```diff
@@ -2,110 +2,37 @@
 // Firefox sucks: `isFinite` is not defined on `window`, see violentmonkey/violentmonkey#300
 // eslint-disable-next-line no-restricted-properties
 export const {
-  console, CustomEvent, Promise, isFinite, Uint8Array,
+  // types
+  Blob, Boolean, Error, Promise, Uint8Array,
+  // props and methods
+  atob, console, isFinite, setTimeout,
 } = global;
 
-const arrayProto = Array.prototype;
-const objectProto = Object.prototype;
-
-const bindThis = func => (thisObj, ...args) => func.apply(thisObj, args);
-
-export const forEach = bindThis(arrayProto.forEach);
-
-export const map = bindThis(arrayProto.map);
-
-export const indexOf = bindThis(arrayProto.indexOf);
-
-export const push = bindThis(arrayProto.push);
-
-export const includes = arrayProto.includes
-  ? bindThis(arrayProto.includes)
-  : (arr, item) => indexOf(arr, item) >= 0;
-
-export const toString = bindThis(objectProto.toString);
-const numberToString = bindThis(Number.prototype.toString);
-const stringSlice = bindThis(String.prototype.slice);
-const stringCharCodeAt = bindThis(String.prototype.charCodeAt);
-const { fromCharCode } = String;
+export const {
+  filter, forEach, indexOf, join, map, push,
+  // arraySlice, // to differentiate from String::slice which we use much more often
+  includes = function _(item) { return this::indexOf(item) >= 0; },
+} = Array.prototype;
 
-export const { keys } = Object;
-export const assign = Object.assign || ((obj, ...args) => {
-  forEach(args, (arg) => {
-    if (arg) {
-      forEach(keys(arg), (key) => {
-        obj[key] = arg[key];
-      });
-    }
-  });
-  return obj;
-});
+export const { keys: objectKeys, defineProperty, defineProperties } = Object;
+export const { charCodeAt, match, slice } = String.prototype;
+export const { toString: objectToString } = Object.prototype;
+const { toString: numberToString } = Number.prototype;
+const { replace } = String.prototype;
+export const { fromCharCode } = String;
+export const { warn } = console;
+export const { addEventListener } = EventTarget.prototype;
+export const { append, setAttribute } = Element.prototype;
+export const { createElement } = Document.prototype;
 
 export const isArray = obj => (
   // ES3 way, not reliable if prototype is modified
-  // toString(obj) === '[object Array]'
+  // Object's toString(obj) === '[object Array]'
```

Comment:
This line is the old code which was deprecated:

```js
toString(obj) === '[object Array]'
```

Maybe it's better to change to
```js
objectToString(obj) === '[object Array]'
// or
Object.prototype.toString.call(obj) === '[object Array]'
```